torchgeo/datasets/hyspecnet.py

```diff
@@ -0,0 +1,229 @@
+
+    To construct HySpecNet-11k, a total of 250 EnMAP tiles acquired during the routine
+    operation phase between 2 November 2022 and 9 November 2022 were considered. The
+    considered tiles are associated with less than 10% cloud and snow cover. The tiles
+    were radiometrically, geometrically and atmospherically corrected (L2A water & land
+    product). Then, the tiles were divided into nonoverlapping image patches. The
+    cropped patches at the borders of the tiles were eliminated. As a result, more than
+    45 patches per tile are obtained, resulting in 11,483 patches for the full dataset.
+
+    We provide predefined splits obtained by randomly dividing HySpecNet into:
+
+    #. a training set that includes 70% of the patches,
+    #. a validation set that includes 20% of the patches, and
+    #. a test set that includes 10% of the patches.
+
+    Depending on the way that we used for splitting the dataset, we define two
+    different splits:
+
+    #. an easy split, where patches from the same tile can be present in different sets
+       (patchwise splitting); and
+    #. a hard split, where all patches from one tile belong to the same set
+       (tilewise splitting).
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2306.00385
+
+    .. versionadded:: 0.7
+    """
+
+    url = 'https://hf.co/datasets/torchgeo/hyspecnet/resolve/13e110422a6925cbac0f11edff610219b9399227/'
+    md5s: ClassVar[dict[str, str]] = {
+        'hyspecnet-11k-01.tar.gz': '974aae9197006727b42ec81796049efe',
+        'hyspecnet-11k-02.tar.gz': 'f80574485f835b8a263b6c64076c0c62',
+        'hyspecnet-11k-03.tar.gz': '6bc1de573f97fa4a75b79719b9270cb3',
+        'hyspecnet-11k-04.tar.gz': '2463dc10653cb8be10d44951307c5e7d',
+        'hyspecnet-11k-05.tar.gz': '16c1bd9e684673e741c0849bd015c988',
+        'hyspecnet-11k-06.tar.gz': '8eef16b67d71af6eb4bc836d294fe3c4',
+        'hyspecnet-11k-07.tar.gz': 'f61f0e7d6b05c861e69026b09130a5d6',
+        'hyspecnet-11k-08.tar.gz': '19d390bc9e61b85e7d765f3077984976',
+        'hyspecnet-11k-09.tar.gz': '197ff47befe5b9de88be5e1321c5ce5d',
+        'hyspecnet-11k-10.tar.gz': '9e674cca126a9d139d6584be148d4bac',
+        'hyspecnet-11k-splits.tar.gz': '94fad9e3c979c612c29a045406247d6c',
+    }
+
+    all_bands = valid_channels_ids
+    rgb_bands = (43, 28, 10)
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        strategy: str = 'easy',
+        bands: Sequence[int] = all_bands,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new HySpecNet11k instance.
+
+        Args:
+            root: Root directory where dataset can be found.
+            split: One of 'train', 'val', or 'test'.
+            strategy: Either 'easy' for patchwise splitting or 'hard' for tilewise
```

Comment:
What is the intuition between "easy" and hard" as the splitting strategy?

Reply:
A great question. I'm just copying the naming used by the dataset authors. It doesn't make a lot of sense to me either.

Reply:
Maybe we can switch that, and change the docs to say "this is referred to as 'easy' " in the original?

Reply:
It would just be extra code. Do you think the current documentation (both in `__init__` and in the class docstring) are insufficient?

Reply:
It's well explained there, just seems unintuitive to call it "easy" and "hard" but fine to keep it then with the orig code.